### PR TITLE
Admin Router: Fix test harnes file descriptors leak

### PR DESCRIPTION
**Pull request by vespian**
## High Level Description

**CI failures are expected, please ignore. This will be addressed when merging the intermediate branch (prozlach/ar-next-5) to master. Same goes for upstream bump - I have done EE tests manually using AR tests, proper integration tests will be done while shipping AR-NEXT branches.**

Please check Jira issues for information on the problem solved here, the docstring of `_event_processing` function for the description of the solution.

## Related Issues
https://jira.mesosphere.com/browse/DCOS_OSS-1164 `Admin Router: fix file descriptors leak, bump pytest to >=3.1.0`

## Related PRs
EE DC/OS PR: https://github.com/mesosphere/dcos-enterprise/pull/1401
Open DC/OS AR-NEXT PR: https://github.com/dcos/dcos/pull/1919

### CODE REVIEW

**Review by mhrabovcin — APPROVED** [review]

File: packages/adminrouter/extra/src/test-harness/modules/runner/common.py [hunk 1]
```diff
@@ -256,38 +244,99 @@ class LogCatcher:
...
+        adding the new LogWriter in its place. Bulletpoint a) guarantees that
+        all these operations will be done in predefined order.
+
+         poll() call generally seems to be easier to use and better fits
+         our use case than e.g. plain select() (no need for global lock
+         while updating FD lists):
         * http://stackoverflow.com/a/25249958
         * http://www.greenend.org.uk/rjk/tech/poll.html
         """
         while True:
```
Comment (gpaul) [inline packages/adminrouter/extra/src/test-harness/modules/runner/common.py:289]: I'm really surprised that we need to build our own polling in a language as high-level as python. Is there no simpler scheme? I would think that there would be some very elegant callback-based or futures-based library that would make this function super simple.

**Review by gpaul — APPROVED** [review]
I wish this was all so much simpler - this seems pretty low-level for something that just aggregates log lines.

File: packages/adminrouter/extra/src/test-harness/modules/runner/common.py (hunk 1 above)
Comment (vespian) [reply packages/adminrouter/extra/src/test-harness/modules/runner/common.py:289]: Using `pool()` seemed simple at the very beginning of test harness, but in time the code grew, features were added, bugs were fixed and now it is the way it is :)

I can definitely look into `asyncio`&friends next time I touch this part of the code!

**Comment by vespian** [comment]
Admin Router: Fix test harnes file descriptors leak

**Pull request merged**